custom_components/mqtt_vacuum_camera/utils/connection/decompress.py

```
@@ -0,0 +1,83 @@
+from isal import igzip, isal_zlib  # pylint: disable=I1101
+from valetudo_map_parser.config.rand25_parser import RRMapParser
+
+from custom_components.mqtt_vacuum_camera.const import LOGGER
+from custom_components.mqtt_vacuum_camera.utils.thread_pool import ThreadPoolManager
+
+
+def _safe_zlib_decompress(data: bytes) -> str:
+    try:
+        return isal_zlib.decompress(data).decode()
+    except Exception as e:
+        raise ValueError(f"Invalid Hypfer payload: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)